src/audio/module_adapter/module/waves/waves.c

```diff
@@ -653,6 +666,7 @@ static int waves_codec_init(struct processing_module *mod)
 			 sizeof(struct waves_codec_data));
 		ret = -ENOMEM;
 	} else {
+		memset(waves_codec, 0, sizeof(struct waves_codec_data));
```

Comment:
why is this needed? Is this fixing a bug - i.e. is any code relying on this being all zeroes? Existing or new code from this commit?

Reply:
@barry-waves sorry, I still don't understand, what the issue was? Why does it have to 0-initialised?

Reply:
Hi @lyakh 
If the waves_codec->config_blob pointer is not NULL, we assume that we have allocated memory to this pointer. We would apply the content where the pointer points to, and we return error since it has invalid data.
So, we need to clean the buffer which gets from module_allocate_memory() function.